client/components/BudgetGrid/EntryFormRow.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component, PropTypes } from 'react';
+
+class EntryFormRow extends Component {
+  static propTypes = {
+    categories: PropTypes.object.isRequired
+  }
+
+  state = {
+    categoryId: 16,
+    description: '',
+    amount: ''
+  }
+
+  getCategoryOptions() {
+    const { categories } = this.props;
+
+    return Object.keys(categories).map(categoryId => (
+      <option
+        key={categoryId}
+        value={categoryId}
+      >
+        { categories[categoryId] }
+      </option>
+    ));
+  }
+
+  handleFieldChange = e => this.setState({ [e.target.name]: e.target.value })
```

Comment:
I'm not sure how to proceed with the `preventDefault`, do you have any resources I could look at regarding that?